song_match/song_match.py

```diff
@@ -66,18 +64,21 @@ def __init__(self, num_players: int = 1):
     async def __init_game_loop(self) -> None:
         current_position = STARTING_POSITION
         while self._song.is_not_finished(current_position, self._game_length):
+            await self.__tap_guard(lambda: self.__play_round_transition_effect())
+            
             notes = self._song.get_sequence_slice(current_position)
             await self.__tap_guard(lambda: self.__play_notes(notes))
 
             await self.__wait_for_players_to_match_notes(current_position)
+            await self.__check_for_game_over()
 
             await sleep(TIME_IN_BETWEEN_PLAYER_AND_COZMO)
 
             await self.__wait_for_cozmo_to_match_notes(current_position)
-
-            await self.__tap_guard(lambda: self.__play_round_transition_effect())
+            await self.__check_for_game_over()
```

Comment:
I don't think these 3 game over checks are necessary.

If you look in `wait_for_players_to_match_notes` and `wait_for_cozmo_to_match_notes` we already call `check_for_game_over`.

Those 2 checks should handle it so you don't need to call it after the loop too.